src/wh_server_crypto.c

```diff
@@ -1963,6 +2007,348 @@ static int _HandleSha256Dma(whServerContext* server, whPacket* packet,
 }
 #endif /* ! NO_SHA256 */
 
+
+#if defined(HAVE_DILITHIUM)
+
+static int _HandleMlDsaKeyGenDma(whServerContext* server, whPacket* packet,
+                                 uint16_t* size)
+{
+#ifdef WOLFSSL_DILITHIUM_NO_MAKE_KEY
+    (void)server;
+    (void)packet;
+    (void)size;
+    return WH_ERROR_NOHANDLER;
+#else
+    int      ret = WH_ERROR_OK;
+    MlDsaKey key[1];
+    void*    clientOutAddr = NULL;
+    uint16_t keySize       = 0;
+
+#if WH_DMA_IS_32BIT
+    wh_Packet_pq_mldsa_keygen_Dma32_req* req = &packet->pqMldsaKeygenDma32Req;
+    wh_Packet_pq_mldsa_Dma32_res*        res = &packet->pqMldsaDma32Res;
+#else
+    wh_Packet_pq_mldsa_keygen_Dma64_req* req = &packet->pqMldsaKeygenDma64Req;
+    wh_Packet_pq_mldsa_Dma64_res*        res = &packet->pqMldsaDma64Res;
+#endif
+
+    /* Check the ML-DSA security level is valid and supported */
+    if (0 == _IsMlDsaLevelSupported(req->level)) {
+        ret = WH_ERROR_BADARGS;
```

Comment:
It's ok to return(BAD_ARGS) here instead of wait til the end.  No change necessary.  Just trying to save an indent